warp/yul/ToCairoVisitor.py

```diff
@@ -1,114 +1,209 @@
+        self.preamble: bool = False
         self.n_names: int = 0
+        self.imports = defaultdict(set)
+        merge_imports(self.imports, COMMON_IMPORTS)
+        self.cairo_code: str = ""
 
     def translate(self, node: ast.Node) -> str:
         main_part = self.print(node)
         return main_part
 
     def print(self, node: ast.Node, *args, **kwargs) -> str:
-        self.visit(node, *args, **kwargs)
-        return self.repr_stack.pop()
+        return self.visit(node, *args, **kwargs)
 
     def common_visit(self, node: ast.Node, *args, **kwargs):
         raise AssertionError(
             f"Each node type should have a custom visit, but {type(node)} doesn't"
         )
 
-    def visit_typed_name(self, node: ast.TypedName):
-        self.repr_stack.append(f"{node.name} : {node.type}")
+    def visit_typed_name(self, node: ast.TypedName) -> str:
+        return f"{node.name} : {node.type}"
 
-    def visit_literal(self, node: ast.Literal):
+    def visit_literal(self, node: ast.Literal) -> str:
         v = int(node.value)  # to convert bools: True -> 1, False -> 0
         high, low = divmod(v, UINT128_BOUND)
-        self.repr_stack.append(f"Uint256(low={low}, high={high})")
-
-    def visit_identifier(self, node: ast.Identifier):
-        self.repr_stack.append(f"{node.name}")
-
-    def visit_assignment(self, node: ast.Assignment):
-        ids_repr = ", ".join("local " + self.print(x) for x in node.variable_names)
-        value_repr = self.print(node.value)
+        return f"Uint256(low={low}, high={high})"
+
+    def visit_identifier(self, node: ast.Identifier) -> str:
+        return f"{node.name}"
+
+    def generate_ids_typed(self, variable_names, function_name: str = ""):
+        variables = []
+        for var in variable_names:
+            var_repr = self.print(var)
+            if "Uint256" not in var_repr:
+                var_repr += ": Uint256"
+            if function_name in RETURNS_TWO:
+                var_repr += ", _"
+            variables.append(var_repr)
+        return ", ".join("local " + x for x in variables)
+
+    def visit_assignment(self, node: ast.Assignment) -> str:
+        self.preamble = False
+        value_repr: str = self.print(node.value)
+        function_name: str = value_repr[: value_repr.find("(")].strip()
+        ids_repr: str = self.generate_ids_typed(node.variable_names, function_name)
         if isinstance(node.value, ast.FunctionCall):
-            if not node.variable_names:
-                self.repr_stack.append(value_repr)
+            function_args: str = value_repr[
+                value_repr.find("(") + 1 : value_repr.rfind(")")
+            ]
+            if function_name in YUL_BUILTINS_MAP.keys():
+                builtin_to_cairo = YUL_BUILTINS_MAP[function_name](function_args)
+                merge_imports(self.imports, builtin_to_cairo.required_imports())
+                if not node.variable_names:
+                    self.preamble = True
```

Comment:
`self.preamble` seems to be set to `True` in all branches of the outer-most if. Can it be moved up and set to True once?